OpenRA.Mods.Common/Traits/Harvester.cs

```diff
@@ -124,9 +126,15 @@ protected override void Created(Actor self)
 		public bool IsEmpty => storesResources.All(sr => sr.ContentsSum == 0);
 		public int Fullness => storesResources.Sum(sr => sr.ContentsSum * 100 / sr.Capacity) / storesResources.Length;
 
-		protected override bool CanLink()
+		public override bool IsLinkingPossible(BitSet<LinkType> type, bool forceEnter = false)
 		{
-			return !IsEmpty;
+			return base.IsLinkingPossible(type, forceEnter) && (forceEnter || !IsEmpty);
+		}
+
+		public override bool CanLinkTo(Actor hostActor, ILinkHost host, bool forceEnter = false, bool ignoreOccupancy = false)
+		{
+			return base.CanLinkTo(hostActor, host, forceEnter, ignoreOccupancy)
+				&& (self.Owner == hostActor.Owner || (ignoreOccupancy && self.Owner.IsAlliedWith(hostActor.Owner)));
```

Comment:
Wouldn't it make more sense to require force enter to unload at an allied refinery?

Reply:
why? Imo you should be able just directly order the harvester. IgnoreOccupancy is used for manual orders